BismarkUnchained/scenes/Player/Player.gd

```diff
@@ -60,14 +65,54 @@ func move_translate(delta):
 
 func move_rotate():
 	if controller:
-		var direction = Vector2(Input.get_joy_axis(device, 2), Input.get_joy_axis(device, 3))
+		var direction = Vector2(Input.get_joy_axis(device, 3), Input.get_joy_axis(device, 4))
 		if direction.length() > 0.5:
+			last_dir = Vector2(direction.x, -direction.y)
+			print(last_dir)
 			rotate_dir(direction)
 	else:
 		var direction = get_global_mouse_position() - get_global_position()
+		last_dir = Vector2(direction.x, -direction.y)
 		rotate_dir(direction)
 		
 
+func shoot():
+	if Input.is_action_pressed("player_shoot"):
+		if get_node("/root/Main").time_elapsed > time_last_shot + time_to_shoot:
```

Comment:
optional: do we want to getter/setter `time_elapsed`? Also why do we need a global `time_elapsed`?